src/materials/czm/CZMVolumetricStrain.C

```diff
@@ -9,176 +9,86 @@ InputParameters CZMVolumetricStrain::validParams() {
-      _disp_vars(3), _disp_old(3), _disp_neighbor_old(3),
-      _czm_total_strain_rate(
-          declareProperty<RankTwoTensor>("czm_total_strain_rate")),
-      _czm_normal_strain_rate(
-          declareProperty<RankTwoTensor>("czm_normal_strain_rate")),
-      _czm_sliding_strain_rate(
-          declareProperty<RankTwoTensor>("czm_sliding_strain_rate")),
-      _ld(_ndisp == 1 ? false : getParam<bool>("large_kinematics")) {
-
-  // Enforce consistency
-  if (_ndisp != _mesh.dimension())
-    paramError("displacements",
-               "Number of displacements must match problem dimension.");
-
-  // enforce  mataterial formulation
-  if (getParam<bool>("use_displaced_mesh"))
-    mooseError("CZMVolumetricStrain doesn't work with use_displaced_mesh true");
-}
-
-void CZMVolumetricStrain::initialSetup() {
-  // initializing the displacement vectors
-  for (unsigned int i = 0; i < _ndisp; ++i) {
-    _disp[i] = &coupledValue("displacements", i);
-    _disp_neighbor[i] = &coupledNeighborValue("displacements", i);
-    _disp_vars[i] = getVar("displacements", i);
-    _disp_old[i] = &coupledValueOld("displacements", i);
-    _disp_neighbor_old[i] = &coupledNeighborValueOld("displacements", i);
-
-    _grad_disp.push_back(&coupledGradient("displacements", i));
-    _grad_disp_neighbor.push_back(&coupledNeighborGradient("displacements", i));
-    _grad_disp_old.push_back(&coupledGradientOld("displacements", i));
-    _grad_disp_neighbor_old.push_back(
-        &coupledNeighborGradientOld("displacements", i));
-  }
-
-  // All others zero (so this will work naturally for 2D and 1D problems)
-  for (unsigned int i = _ndisp; i < 3; i++) {
-    _disp[i] = &_zero;
-    _disp_neighbor[i] = &_zero;
-    _disp_old[i] = &_zero;
-    _disp_neighbor_old[i] = &_zero;
-    _grad_disp.push_back(&_grad_zero);
-    _grad_disp_neighbor.push_back(&_grad_zero);
-    _grad_disp_old.push_back(&_grad_zero);
-    _grad_disp_neighbor_old.push_back(&_grad_zero);
-  }
+      _base_name(isParamValid("base_name") &&
+                         !getParam<std::string>("base_name").empty()
+                     ? getParam<std::string>("base_name") + "_"
+                     : ""),
+      _displacement_jump_global(getMaterialPropertyByName<RealVectorValue>(
+          _base_name + "displacement_jump_global")),
+      _czm_total_rotation(getMaterialPropertyByName<RankTwoTensor>(
+          _base_name + "czm_total_rotation")),
+      _czm_total_strain(declarePropertyByName<RankTwoTensor>(
+          _base_name + "czm_total_strain")),
+      _czm_normal_strain(declarePropertyByName<RankTwoTensor>(
+          _base_name + "czm_normal_strain")),
+      _czm_sliding_strain(declarePropertyByName<RankTwoTensor>(
+          _base_name + "czm_sliding_strain")),
+      _dadA_mp(declarePropertyByName<Real>(_base_name + "czm_area_ratio")),
+      // _czm_total_strain_old(getMaterialPropertyOldByName<RankTwoTensor>(
```

Comment:
Why are these commented out?

Reply:
we don't need these extra material properties, rates will be computed using post processor, faster and less memory usage.